src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/specification/UserSpecification.kt

```diff
@@ -151,14 +164,24 @@ class UserSpecification(private val filter: UserFilter) : Specification<UserPers
     }
 
     filter.activeCaseloadId?.run {
-      predicates.add(activeCaseload(this))
+      if (filter.showOnlyLSAs == false) {
+        predicates.add(activeCaseload(this))
+      }
+    }
+
+    if (filter.showOnlyLSAs == true) {
+      filter.activeCaseloadId?.run {
+        predicates.add(localAuthorityOfAdminGroup(filter.activeCaseloadId))
+      } ?: lsaOnly()
```

Comment:
Doesn't this need to be `predicates.add(lsaOnly())`

Reply:
this is directly adding to user defined 'join' function to the 'root:Root<UserPersonDetail>'